tracing/src/main/java/org/axonframework/extensions/tracing/TracingCommandGateway.java

```diff
@@ -7,6 +38,11 @@
 
 import java.util.concurrent.CompletableFuture;
 
+/**
+ * A tracing command gateway which activates a calling {@link Span}, when the {@link CompletableFuture} completes.
+ *
+ * @author Christophe Bouhier
+ */
 public class TracingCommandGateway extends DefaultCommandGateway {
```

Comment:
Would it make sense to promote the tracing dispatch and handler interceptors to a global level here?
That would make enforcing those to be a part of the `TracingCommandGateway` easier too, as well as testing their existence in the auto config test.
What do you think?

Reply:
Can you elaborate? I am not sure what you mean by 'promote to global level'?

Reply:
I thought it might make sense to do this:
```
public class TracingCommandGateway extends DefaultCommandGateway {

public class TracingCommandGateway extends DefaultCommandGateway {

    private final Tracer tracer;
    private final OpenTraceDispatchInterceptor openTraceDispatchInterceptor;
    private final OpenTraceHandlerInterceptor openTraceHandlerInterceptor;

    public TracingCommandGateway(Builder builder, Tracer tracer) {
        super(builder);
        this.tracer = builder.tracer;
        this.openTraceDispatchInterceptor = builder.openTraceDispatchInterceptor;
        this.openTraceHandlerInterceptor = builder.openTraceHandlerInterceptor;
    }
...
}
```

Additionally, if you'd provide a `TracingCommandGateway.Builder` class which contains those three fields, the `Builder#validate()` function can ensure you that there will __always be__ a handler interceptor, dispatcher interceptor and a tracer for this `TracingCommandGateway`, thus eliminating the need to verify that in the spring boot auto configure test.

I'd be happy to do a PR for this minor change though, would you be okay with that @dzonekl?

Reply:
ah ok @smcvb yes, this would be cleaner. Please go ahead with this change, thank you!

Reply:
Cool, I'll see if I can find the time for that today! :)

Reply:
FYI: I think I'll do this after this PR has been merged, so not to hit potential merge conflicts.